src/jenkins_job_insight/storage.py

```
@@ -478,6 +667,722 @@ def _count_child_failures_recursive(child: dict) -> int:
+        classifications = {}
+        for row in await cursor.fetchall():
+            if row[0]:
+                classifications[row[0]] = row[1]
+
+        # Recent runs (failures only, since we only track failures)
+        cursor = await db.execute(
+            f"""SELECT job_id, job_name, build_number, error_message, error_signature,
+                       classification, child_job_name, child_build_number, analyzed_at
+                FROM failure_history WHERE test_name = ?{job_filter}
+                ORDER BY analyzed_at DESC LIMIT ?""",
+            params + [limit],
```

Ruff (0.15.6)

[warning] Consider `[*params, limit]` instead of concatenation

Replace with `[*params, limit]`

(RUF005)